pypsa-to-iamc

```diff
@@ -0,0 +1,634 @@
+
+            #Emissions : Oil 
+
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Electricity|Liquids|Fossil'] = 0.001*ct*(n.links_t.p0.filter(like ='oil-').filter(like =country)).sum().sum()
+            
+                
+            ### Heat  (MW -> GW)
+            
+            #Emissions :Biomass (CCS)
+
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Biomass|CCS'] = 0.001*ct*(n.links_t.p2.filter(like ='solid biomass CHP CC').filter(like =country)).sum().sum()
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Solid bio and waste|CCS'] = var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Biomass|CCS']  
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Solid bio and waste|Primary solid biomass|CCS'] = var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Solid bio and waste|CCS']  
+
+
+            #Emissions :Natural gas
+
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Gases|Fossil|Natural Gas'] = 0.001*ct*(n.links_t.p3.filter(like ='gas CHP').filter(like =country)).sum().sum()          
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Gases|Fossil|Natural Gas|CCS'] = 0.001*ct*(n.links_t.p3.filter(like ='gas CHP CC').filter(like =country)).sum().sum()                       
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Gases|Fossil|Natural Gas'] += 0.001*ct* (n.links_t.p2.filter(like ='gas boiler').filter(like =country)).sum().sum()
+            
+            #Emissions : Oil
+ 
+            var[sh]['Emissions|Kyoto gases|Fossil|CO2|Heat|Liquids|Fossil'] = 0.001*ct*(n.links_t.p2.filter(like ='oil boiler').filter(like =country)).sum().sum() 
+            
+
+            
+            ###Energy generation: total,peaks,and percentiles
+            
+            sheet_var={"Yearly_generation_supply":'Generation|Yearly',"Winter_peak_generation":'Generation|Winter peak',"summer_peak_generation":'Generation|Summer peak',\
+                       "Percentile50_generation":'Generation|Percentile 50',"Percentile25_generation":'Generation|Percentile 25',\
+                       'Percentile_25_consumpt':'Hourly power consumption|Percentile 25','Percentile_50_consumpt':'Hourly power consumption|Percentile 50','Summer_peak_consumpt' :'Hourly power consumption|Summer peak',\
+                       'Winter_peak_consumpt':'Hourly power consumption|Winter peak'}
+            
+            ct=3        # Hourly resolution 
+        
+            for sheet in sheets.keys(): 
+             if sheet in ("Yearly_generation_supply","Winter_peak_generation","summer_peak_generation","Percentile50_generation","Percentile25_generation"):
+              sh=sheets[sheet]
+               
+              #Energy generation : Solar PV (rooftop, ground)   
+            
+              var[sh][sheet_var[sheet]+'|Electricity|Solar|Rooftop PV'] = 0.001*ct*gen_des(time(n.generators_t.p.filter(like ='solar rooftop').filter(like =country).sum(axis=1)))
```

Comment:
update here and below:
*ct
*name of function gen_des()
*name of function time()

Reply:
Done